model/TaskLog/CategoryCollectionDecorator.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2017 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+namespace oat\taoTaskQueue\model\TaskLog;
+
+use oat\taoTaskQueue\model\Entity\CategoryEntityDecorator;
+use oat\taoTaskQueue\model\TaskLogInterface;
+
+/**
+ * Interface CategoryCollectionDecorator
+ *
+ * @author Gyula Szucs <gyula@taotesting.com>
+ */
+class CategoryCollectionDecorator extends TaskLogCollectionDecorator
```

Comment:
Very nice approach with decorating stuff.
I think the name it's not truly respecting the truth :). In my opinion it should be something like ConfigCollectionDecorator, ConfigEntityDecorator, because we are decorating with stuff from config (in our case it's the category) but if we want to decide to show task association infos (which it's another config) we will need to add a new decorator. 
What I am trying to say it's quite to specific, maybe only rename the ConfigCollectionDecorator, because the CategoryEntityDecorator can make sense, but for a new config to display we will need to create another collection decorator.

Just my thought on seeing this, what do you think?

Reply:
Yes, I agree, the name is quite specific and I would be amenable to change it but I would say being specific is the point. Because CategoryCollectionDecorator or CategoryEntityDecorator specifically tell me that those classes are responsible to put the category in the result. But if used ConfigCollectionDecorator (of course, using that, we could spare some time not creating new decorators every time) I cannot be really sure what kind of config is included in the result. I should have to go in the class and check what is does. I would say both ways can be acceptable.